### PR TITLE
Get all query

### PR DESCRIPTION
**Pull request by jdhiguita**
## Summary
- Add method in book repository with query custom
- Add test method to new method in book repository integration test
- Add method in book controller to find by all fields filters
## Trello Card
https://trello.com/c/3ov6klra

### CODE REVIEW

File: src/main/java/wolox/training/services/impl/BookServiceImpl.java [hunk 1]
```diff
@@ -66,11 +67,17 @@ public Book createByOpenLibrary(String isbn) {
         }
     }
 
+    @Override
+    public List<Book> findByAllFields(String genre, String author, String image, String title,
```
Comment (cramallo) [inline src/main/java/wolox/training/services/impl/BookServiceImpl.java]: it's okay. I leave you only a comment, for clean code purposes you could recieve here a Request model class as parameter. This class could be built in the controller.

Comment (jdhiguita) [reply src/main/java/wolox/training/services/impl/BookServiceImpl.java]: Add Request model as parameter for the method

**Review by cramallo — APPROVED** [review]

File: src/main/java/wolox/training/services/impl/BookServiceImpl.java [hunk 2]
```diff
@@ -66,11 +68,20 @@ public Book createByOpenLibrary(String isbn) {
         }
     }
 
+    @Override
+    public List<Book> findByAllFields(Model model) {
```
Comment (cramallo) [inline src/main/java/wolox/training/services/impl/BookServiceImpl.java]: Model is very generic you could create a model request more representative

**Review by cramallo — CHANGES_REQUESTED** [review]

File: src/main/java/wolox/training/services/impl/BookServiceImpl.java (hunk 2 above)
Comment (jdhiguita) [reply src/main/java/wolox/training/services/impl/BookServiceImpl.java]: Create a BookParametersDTO for all fields parameters

**Review by cramallo — APPROVED** [review]

**Review by Gianni-Nicolas — APPROVED** [review]

**Pull request merged**